protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -122,6 +125,18 @@ definitions:
           - SOURCE_RETRIEVAL_ERROR
           # Errors casting to appropriate type
           - DESTINATION_TYPECAST_ERROR
+  AirbyteRecordMessageFileReference:
+    type: object
+    additionalProperties: true
+    properties:
+      local_path:
+        type: string
+        description: |-
+          The path to the referenced file on the local shared volume.
+      file_size_bytes:
+        type: integer
+        description: |-
+          The size of the referenced file in bytes.
```

Comment:
Nit/Question: If these file references will always be part of a stream that describes the file in more detail (original URI, mime-type, etc) wouldn't  `file_size_bytes` be in the `record.data` json?  Is that information duplicated here so that the destination and/or platform can decide if the file is too big (or small?) to deal with?

Reply:
Good question: these top level fields are for internal airbyte consumption. This one specific would be consumed by the orchestrator to be tallied for billing purposes. The `record.data` may also contain this information, but I do not believe we want to be introspecting data fields (which could theoretically be de-selected / filtered) for internal book-keeping.

Reply:
Note that the current source-s3 implementation return `{"file_url": absolute_file_path, "bytes": file_size, "file_relative_path": file_relative_path}`. Do we need the relative path?

Reply:
good question, I can add both.

Reply:
@maxi297 do you have a link to where we define the currently used properties today?

Reply:
It is a bit spread as it is defined in each source but here is the example for source-s3: https://github.com/airbytehq/airbyte/blob/ab088457429302b11d8909cad90631ad0f2a9f64/airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py#L261. 

We also add some information as part of the CDK: https://github.com/airbytehq/airbyte-python-cdk/blob/837913f6372a6465472fae269dac7042f336945f/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py#L148-L154

This is being assembled as a FileTransferMessage here: https://github.com/airbytehq/airbyte-python-cdk/blob/837913f6372a6465472fae269dac7042f336945f/airbyte_cdk/sources/utils/record_helper.py#L39-L42

Reply:
I see
```
                "file_url" to fileMessage.fileUrl,
                "file_relative_path" to fileMessage.fileRelativePath,
                "source_file_url" to fileMessage.sourceFileUrl,
                "modified" to fileMessage.modified,
                "bytes" to fileMessage.bytes,
```

in the dest codebase, of which only `file_relative_path` and `file_relative_path`. So I'm planning on only adding those for now.